File: ProblemSet_33xx/Problem_3315/solution.cs
public class Solution {
    public int[] MinBitwiseArray(IList<int> nums) {
        int[] result = new int[nums.Count];
        for (int i = 0; i < nums.Count; i++)
        {
            int countOf1 = 0;
            int temp = nums[i];
            while (temp % 2 == 1)
            {
                temp >>= 1;
                countOf1++;
            }

            if (countOf1 == 0)
            {
                result[i] = -1;
                continue;    
            }

            int pow2 = 1 << (countOf1 - 1);
            result[i] = (nums[i] ^ pow2) | (pow2 - 1);
        }
        return result;
    }
}

File: ProblemSet_33xx/Problem_3315/solution.py
from typing import List


class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        from typing import List


class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        def solveBitwise(num: int) -> int:
            if (num % 2 == 0):
                return -1
            num_str = str(bin(num))
            
            result = []
            flag = False
            for i in range(len(num_str)-1, 1, -1):
                if (num_str[i] == "1" and flag == False and (i == 2 or  num_str[i-1] == "0")):
                    result.append("0")
                    flag = True
                else:
                    result.append(num_str[i])
            result_str = "".join(result)[::-1]
            return int(result_str, 2)
        
        return [solveBitwise(num) for num in nums]